python_modules/libraries/dagster-aws/dagster_aws/pipes/clients/ecs.py

```diff
@@ -134,47 +148,95 @@ def run(
                     }
                 )
 
-            params["overrides"] = (  # pyright: ignore (reportGeneralTypeIssues)
+            run_task_params["overrides"] = (  # pyright: ignore (reportGeneralTypeIssues)
                 overrides  # assign in case overrides was created here as an empty dict
             )
 
-            response = self._client.run_task(**params)
+            response = self._client.run_task(**run_task_params)
+
+            if len(response["tasks"]) > 1:
+                # this error should never happen, as we're running a single task
+                raise DagsterInvariantViolationError(
+                    f"Expected to get a single task from response, got multiple: {response['tasks']}"
+                )
 
-            tasks: List[str] = [task["taskArn"] for task in response["tasks"]]  # pyright: ignore (reportTypedDictNotRequiredAccess)
+            task = response["tasks"][0]
+            task_id = task["taskArn"].split("/")[-1]  # pyright: ignore (reportTypedDictNotRequiredAccess)
+
+            def get_cloudwatch_params(container_name: str) -> Optional[Dict[str, str]]:
+                """This will either return the log group and stream for the container, or None in case of a bad log configuration."""
+                if log_config := log_configurations.get(container_name):
+                    if log_config["logDriver"] == "awslogs":
+                        log_group = log_config["options"]["awslogs-group"]  # pyright: ignore (reportTypedDictNotRequiredAccess)
+
+                        # stream name is combined from: prefix, container name, task id
+                        log_stream = f"{log_config['options']['awslogs-stream-prefix']}/{container_name}/{task_id}"  # pyright: ignore (reportTypedDictNotRequiredAccess)
+
+                        return {"log_group": log_group, "log_stream": log_stream}
+                    else:
+                        context.log.warning(
+                            f"[pipes] Unsupported log driver {log_config['logDriver']} for Pipes container {container_name} in task {task['taskArn']}. Dagster Pipes won't be able to read CloudWatch logs from this container."
+                        )
+                else:
+                    context.log.warning(
+                        f"[pipes] log configuration for container {container_name} not found in task definition {task_definition}."
+                    )
+                return None
 
             try:
-                response = self._wait_for_tasks_completion(tasks=tasks, cluster=cluster)
+                if (
```

Comment:
nit: consolidate access to `task["containers"]` up top and only have one pyright warning